db/migrate/20210616133601_create_report_keywords.rb

```diff
@@ -0,0 +1,15 @@
+# rubocop:disable Rails/CreateTableWithTimestamps
+class CreateReportKeywords < ActiveRecord::Migration[6.0]
+  def change
+    create_table :report_keywords do |t|
+      t.string :name, null: false, index: { unique: true }
+    end
+
+    create_table :host_reports_report_keywords, id: false do |t|
+      t.belongs_to :host_report
+      t.belongs_to :report_keyword
+      t.index %i[host_report_id report_keyword_id], unique: true, name: 'index_on_host_report_report_keyword'
```

Comment:
This is dangerous, you are creating a composite index report-keyword. That will be used by the SQL server when we search for keywords associated with a particular report. But we will also need to search for reports with a particular keyword, the opposite direction and SQL server cannot utilize such an index.

The correct solution is to create two separate indexes on both IDs, that is for our case. There are rare situations where also both composite indexes would be required, but that would be only for low-cardinality values which is not our case.